tests/run.sh

```diff
@@ -160,6 +160,10 @@ if [ -n "$K8S" ]; then
     aws eks update-kubeconfig --region $AWS_DEFAULT_REGION --name $AWS_EKS_CLUSTER_NAME --kubeconfig ${HOME}/kubeconfig.${K8S}
 fi
 if test ${CI_NODE_TOTAL:-1} -gt 1; then
+  # Failsafe check if the python script fails to return any tests to run
+  # https://northerntech.atlassian.net/browse/QA-950
+  python ci-parallel-pytest-plugin.py || exit 1
+
   PYTEST_NODES=$(python ci-parallel-pytest-plugin.py | tr '\n' ' ')
```

Comment:
Maybe the following is better:

```suggestion
  PYTEST_NODES=$(python ci-parallel-pytest-plugin.py | tr '\n' ' ')
  test $? -eq 0 || exit 1
```

Reply:
That won't work: the `test $?` is referring to the `tr '\n' ' '`.
We should set `-o pipefail` in the bash options of this script, to make it really failsafe and clean, but I guess this had not be done for some reason.

Reply:
Aha! What about adding `-o pipefail` here https://github.com/mendersoftware/integration/blob/101408cdeb90c1d8a0f394c4f33c53a52fb89e3d/tests/run.sh#L16 ?